backend/routes/api/spots.js

```diff
@@ -0,0 +1,160 @@
+const { handleValidationErrors } = require('../../utils/validation');
+const { Sequelize } = require('sequelize')
+
+const { setTokenCookie, requireAuth } = require('../../utils/auth');
+const { Spot } = require('../../db/models');
+const { SpotImage } = require('../../db/models');
+const { User } = require('../../db/models');
+const { Review } = require('../../db/models');
+
+
+const router = express.Router();
+
+const avgStarRating = (reviews) => {
+
+}
+
+const validateSpot = [
+    check('address')
+        .exists({ checkFalsy: true})
+        .withMessage('Street address is required'),
+    check('city')
+        .exists({ checkFalsy: true})
+        .withMessage('City is required'),
+    check('state')
+        .exists({ checkFalsy: true})
+        .withMessage('State is required'),
+    check('country')
+        .exists({ checkFalsy: true})
+        .withMessage('Country is required'),
+    check('lat')
+        .exists({ checkFalsy: true})
+        .withMessage('Latitude must be within -90 and 90'),
+    check('lng')
+        .exists({ checkFalsy: true})
+        .withMessage('Longitude must be within -180 and 180'),
+    check('name')
+        .exists({ checkFalsy: true})
+        .withMessage('Name must be less than 50 characters'),
+    check('description')
+        .exists({ checkFalsy: true})
+        .withMessage('Description is required'),
+    check('price')
+        .exists({ checkFalsy: true})
+        .withMessage('Price per day must be a positive number'),
+    handleValidationErrors
+];
+
+
+router.get('/:spotId', async (req, res) => {
+    const spotid = parseInt(req.params.spotId);
+    
+    
+   const spot = await Spot.findOne({
+        where: {id: spotid},
+        attributes: {
+            include: [
+              [
+                Sequelize.fn('COUNT', Sequelize.col('Reviews.id')),
+                'numReviews',
+              ],
+              [
+                Sequelize.fn('AVG', Sequelize.col('Reviews.stars')),
+                'avgStarRating',
+              ],
+            ],
+          },
+        include: [
+            {
+                model: Review,
+                attributes: []
+            },
+            {
+                model: SpotImage,
+                attributes: ['id', 'url', 'preview']
+            },
+            {
+                model: User,
+                as: 'Owner',
+                attributes: ['id', 'firstName', 'lastName']
+            }
+        ], group: ['Spot.id', 'SpotImages.id', 'Owner.id']
+   })
+
+
+
+
+
+    return res.json(spot);
+})
+
+router.put('/:spotId', requireAuth, validateSpot, async (req, res) => {
+    if (req.user.id !== req.params.spotId) {
+        return new Error('You do not own this spot.')
+    }
+
+    const {
+         address, city, state, country, 
+         lat, lng, name, description, price
+        } = req.body
+
+    const spot = await Spot.findOne({
+        where: {
+            id: req.params.spotId
+        }
+    })
+
+    spot.update()
```

Comment:
These values aren't being saved. They need to be passed into the update function:
https://sequelize.org/docs/v7/querying/update/